### PR TITLE
feat: add prometheus metrics

### PR DESCRIPTION
**Pull request by mrlihanbo**

### CODE REVIEW

File: pkg/stats/stats.go [hunk 1]
```diff
@@ -60,44 +59,32 @@ func convertTToFields(tags []Tag) log.Fields {
...
-	return nil
 }
 
-func (s *mockMetrics) Counter(name string, val interface{}, tags ...Tag) error {
+func (s *mockMetrics) Counter(name string, val interface{}, tags ...Tag) {
 	if s.logMetrics {
 		fields := convertTToFields(tags)
-		msg := fmt.Sprintf("counter %s.%s + %v", s.prefix, name, val)
+		msg := fmt.Sprintf("rate %s_%s + %v", s.prefix, name, val)
 		log.WithFields(fields).Infoln(msg)
```
Comment (gary-lgy) [inline pkg/stats/stats.go:74]: Should we consider removing the logrus dependency?

File: pkg/controllers/follower/controller.go [hunk 2]
```diff
@@ -234,13 +234,18 @@ func (c *Controller) reconcileLeader(
...
 ) (status worker.Result) {
-	c.metrics.Rate(fmt.Sprintf("follower-controller-%s.throughput", key.sourceGK.String()), 1)
+	c.metrics.Counter("follower_controller_throughput", 1)
 	ctx, keyedLogger := logging.InjectLoggerValues(ctx, "origin", "reconcileLeader", "type", key.sourceGK.String(), "key", key.ObjectSourceKey())
 	startTime := time.Now()
 
 	keyedLogger.V(3).Info("Starting reconcileLeader")
 	defer func() {
-		c.metrics.Duration(fmt.Sprintf("follower-controller-%s.latency", key.sourceGK.String()), startTime)
+		c.metrics.Duration(
```
Comment (gary-lgy) [inline pkg/controllers/follower/controller.go:243]: Should we add these KV to all controller reconcile metrics?

File: cmd/controller-manager/app/options/options.go [hunk 3]
```diff
@@ -123,6 +128,16 @@ func (o *Options) AddFlags(flags *pflag.FlagSet, allControllers []string, disabl
 	flags.BoolVar(&o.EnablePodPruning, "enable-pod-pruning", false, "Enable pod pruning for pod informer. "+
 		"Enabling this can reduce memory usage of the pod informer, but will disable pod propagation.")
 	o.addKlogFlags(flags)
+
+	flags.BoolVar(&o.PrometheusMetrics, "export-prometheus", true, "Whether to expose metrics through a prometheus endpoint")
+	flags.StringVar(&o.PrometheusAddr, "prometheus-addr", "", "prometheus collector address")
+	flags.Uint16Var(&o.PrometheusPort, "prometheus-port", 9090, "prometheus collector port")
+	flags.StringToStringVar(
```
Comment (Poor12) [inline cmd/controller-manager/app/options/options.go:135]: Why do we need to pass quantiles as parameters to the controller? Kind of rare.

File: cmd/controller-manager/app/options/options.go [hunk 4]
```diff
@@ -123,6 +128,16 @@ func (o *Options) AddFlags(flags *pflag.FlagSet, allControllers []string, disabl
 	flags.BoolVar(&o.EnablePodPruning, "enable-pod-pruning", false, "Enable pod pruning for pod informer. "+
 		"Enabling this can reduce memory usage of the pod informer, but will disable pod propagation.")
 	o.addKlogFlags(flags)
+
+	flags.BoolVar(&o.PrometheusMetrics, "export-prometheus", true, "Whether to expose metrics through a prometheus endpoint")
+	flags.StringVar(&o.PrometheusAddr, "prometheus-addr", "", "prometheus collector address")
```
Comment (Poor12) [inline cmd/controller-manager/app/options/options.go:133]:
```suggestion
	flags.StringVar(&o.PrometheusAddr, "prometheus-addr", "", "Prometheus collector address")
```
Capitalize the first letter of the description.

**Review by gary-lgy — APPROVED** [review]

**Pull request merged**